compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TaintAnalyzer.java

```diff
@@ -242,11 +248,9 @@ public TaintAnalyzer(CompilerContext context) {
     }
 
     public BLangPackage analyze(BLangPackage pkgNode) {
-        blockedNode = null;
-        taintErrorSet = new LinkedHashSet<>();
+        resetTempState();
```

Comment:
Temporary state? That doesn't feel correct. Whats the purpose of this method?

Reply:
This methods clears the state maintained for each top-level node analysis. Agreed that this seems not the place to do it. Hence, I have mode it to BLangFunction and BLangResource analysis sections, so that the temporary state maintained will be cleared after each analysis.